src/Microsoft.Azure.WebJobs.Host/Common/ITriggerBindingStrategy.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    /// <summary>
+    /// Strategy pattern to describe how to bind a core trigger type to various parameter.
+    /// Suports:
+    /// - both Single-item vs. Batch dispatch.  
+    /// - core binding, string, poco w/ Binding Contracts
+    /// 
+    /// For example, a single EventHubTriggerInput -->  can bind to 
+    ///  EventData, EventData[], string, string[], Poco, Poco[]    
```

Comment:
It uses IConverterManager to do lookups to and from TMessage. 
Although EventHubs is not currently supporting byte[]. I need need to add it...

---
In reply to: [54934220](https://github.com/Azure/azure-webjobs-sdk/pull/643#discussion_r54934220) [](ancestors = 54934220)